src/include/miopen/any_solver.hpp

```diff
@@ -158,6 +188,45 @@ struct AnySolver
             return TestSysDbRecord(record, std::integral_constant<bool, TunableSolver::Is>());
         }
 
+        // tunable legacy solver
+        std::vector<ConvSolution>
+        GetAllSolutions(const ConvolutionContext& ctx, std::true_type, std::true_type) const
+        {
+            std::vector<ConvSolution> solutions;
+            solutions.push_back(value.GetSolution(ctx, value.GetDefaultPerformanceConfig(ctx)));
+            return solutions;
+        }
+
+        // tunable solver, not legacy
+        std::vector<ConvSolution>
+        GetAllSolutions(const ConvolutionContext& ctx, std::true_type, std::false_type) const
+        {
+            return miopen::solver::GetAllSolutions(value, ctx);
+        }
+
+        // non tunable solver
+        std::vector<ConvSolution>
+        GetAllSolutions(const ConvolutionContext& ctx, std::false_type, std::true_type) const
```

Comment:
Let's MIOPEN_THROW and be ensured that unwanted calls do not happen.

Reply:
I found a use case for this. It will be useful for this to return the solution for a non-tunable solver. This will allow gathering of timing information (fdb) at the same time as pdb.

Reply:
> This will allow gathering of timing information (fdb) at the same time as pdb.

But fdb update should happen after **_all tuning_** (tuning of all Solvers) is done. And if you would like to collect both fdb and pdb at once, how would you prevent configs from non-tunable solvers from getting into pdb?

Reply:
Now I see, thanks. Resolved!